### ISSUE TITLE
Using a space in a file name results in 404

### ISSUE THREAD

**Issue by JojokCreator** [withastro/astro#9945]
### Astro Info

```block
Astro                    v4.2.6
Node                     v20.10.0
System                   Windows (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/tailwind
                         @astrojs/react
                         @astrojs/sitemap
                         astro-icon
```

### Describe the Bug

When creating a new Astro page with a space in it the resulting url goes to a 404 page.

### What's the expected result?

IT should go to the new page

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-astro-ukeyxn?file=src%2Fpages%2Findex.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by ktym4a** [comment]
Maybe fixed in #9934.
Can you try with astro@4.3.1?

**Comment by JojokCreator** [comment]
ahh perfect. It's good now.
